libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -212,9 +213,16 @@ class ChatVertexAI(_VertexAICommon, BaseChatModel):
     def validate_environment(cls, values: Dict) -> Dict:
         """Validate that the python package exists in environment."""
         is_gemini = is_gemini_model(values["model_name"])
+        safety_settings = values["safety_settings"]
+
+        if safety_settings and not is_gemini:
+            raise ValueError("Safety settings are only supported for Gemini models")
+
         cls._init_vertexai(values)
         if is_gemini:
-            values["client"] = GenerativeModel(model_name=values["model_name"])
+            values["client"] = GenerativeModel(
+                model_name=values["model_name"], safety_settings=safety_settings
```

Comment:
`safety_filters` can be also passed to `send_message` (and override the initial config) , e.g. here: https://github.com/langchain-ai/langchain/blob/5ca74c784eee0a76c0e22a0d293d9a793fb0f1c9/libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py#L274

so I'd suggest to allow passing `safety_filters` through `kwargs` to `_generate` (and use them :) )

Reply:
added handling safety_settings in _generate/_agenerate. 
it'd been present in initial commit but it disappeared after some changes.